WordPress/Classes/Services/PostService.m

```diff
@@ -804,7 +804,10 @@ - (NSDictionary *)remoteSyncParametersDictionaryForRemote:(nonnull id <PostServi
 - (void)updatePost:(AbstractPost *)post withRemotePost:(RemotePost *)remotePost {
     NSNumber *previousPostID = post.postID;
     post.postID = remotePost.postID;
-    post.author = remotePost.authorDisplayName;
+    // Used to populate author information for self-hosted sites.
+    BlogAuthor *author = [post.blog getAuthorWithId:remotePost.authorID];
+
+    post.author = remotePost.authorDisplayName ?: author.displayName;
```

Comment:
My understanding here is that `remotePost.authorDisplayName` is `nil` for self-hosted sites, so we fall back to using the `Blog.authors` set to retrieve the author display name, is that correct?

If so, do you know if the API is supposed to return the `authorDisplayName` but not doing so?

Reply:
> My understanding here is that remotePost.authorDisplayName is nil for self-hosted sites, so we fall back to using the Blog.authors set to retrieve the author display name, is that correct?

Correct.

> If so, do you know if the API is supposed to return the authorDisplayName but not doing so?

From my understanding, only an author ID is returned from the XML-RPC call ([see the Return Values section for `wp.getPost` here](https://codex.wordpress.org/XML-RPC_WordPress_API/Posts)).

Reply:
I see, thank you for the link 🙇